pdc/apps/package/serializers.py

```diff
@@ -221,3 +222,57 @@ class BuildImageRTTTestsSerializer(StrictSerializerMixin, serializers.ModelSeria
     class Meta:
         model = models.BuildImage
         fields = ('id', 'build_nvr', 'format', 'test_result')
+
+
+class RepositoryField(serializers.SlugRelatedField):
+
+    def __init__(self, **kwargs):
+        super(RepositoryField, self).__init__(slug_field='id', queryset=Repo.objects.all(), **kwargs)
+
+    def to_representation(self, value):
+        export_data = value.export()
+        export_data["id"] = value.id
+        export_data["content_format"] = {"name": value.content_format.name,
+                                         "pdc_endpoint": value.content_format.pdc_endpoint}
+        return export_data
+
+
+class ReleasedFilesSerializer(StrictSerializerMixin, serializers.ModelSerializer):
+    file_primary_key = serializers.IntegerField(allow_null=False)
+    repo = RepositoryField()
+    released_date = serializers.DateField(required=False)
+    release_date = serializers.DateField()
+    created_at = serializers.DateTimeField(required=False, read_only=True)
+    updated_at = serializers.DateTimeField(required=False, read_only=True)
+    zero_day_release = serializers.BooleanField(required=False, default=False)
+    obsolete = serializers.BooleanField(required=False, default=False)
+
+    class Meta:
+        model = models.ReleasedFiles
+        fields = ('id', 'file_primary_key', 'repo', 'released_date',
+                  'release_date', 'created_at', 'updated_at',
+                  'zero_day_release', 'obsolete')
+
+    def validate(self, data):
+        if "repo" in data:
+            repo_format = data["repo"].content_format
+            repo_name = data["repo"].name
+            if str(repo_format) != "rpm":
+                raise serializers.ValidationError(
+                    {'detail': 'Currently we '
+                               'just support rpm type of repo, the type of %s is %s ' % (repo_name, repo_format)})
+
+        # if there are other types support, should add codes check if the primary key in right table
+
+        return data
+
+    def to_representation(self, value):
+        rep_data = super(ReleasedFilesSerializer, self).to_representation(value)
+        if "file_primary_key" in rep_data and "repo" in rep_data:
+            if rep_data["repo"]["content_format"]["name"] == "rpm":
+                d = models.RPM.objects.get(id=rep_data["file_primary_key"])
```

Comment:
This can throw `models.RPM.DoesNotExist`.

Reply:
Do you mean these error? It can return to the response.
{"detail":"Not found:  RPM matching query does not exist."}

Reply:
OK, looks good.